src/vm_cpu.rs

```diff
@@ -338,8 +340,83 @@ impl VmPmuState {
             }
         }
     }
-}
 
+    /// Saves the internal state for PMU counters. Stops started counters, and resets all configured counters.
+    /// This should be called in anticipation of an outbound context switch.
+    pub fn save_counters(&mut self) {
```

Comment:
Does this need to be `pub`? Same with `restore_counters()` -- they're only called from within this module.